electron/launcher/main-process/helpers/update-game-files.js

```diff
@@ -0,0 +1,346 @@
+const request = require('helpers/request-wrapper').request;
+const _request = require('request');
+const crypto = require('crypto');
+const path = require('path');
+const async = require('async');
+const fs = require('fs-extra');
+const qs = require('qs');
+
+
+const app = require('electron').app;
+
+const config = require('config.json').DJANGO_SERVER;
+const djangoUrl = config.HOST +":" +config.PORT;
+
+
+var localHashes, serverHashes, fileDifference;
+
+
+/***
+================
+HELPER FUNCTIONS
+================
+***/
+
+// CALCULATES MD5 CHECKSUM FOR EVERY FILE IN `masterList` RETURNS MAP
+calculateHashes = masterList => {
+  return new Promise((resolve, reject) => {
+    let calculatedList = new Map();
+    let fileList = masterList.keys();
+
+    async.eachSeries(fileList, (file, callback) => {
+      let stream = fs.createReadStream(path.join(gameInstallationPath, file));
+      let hash = crypto.createHash('md5');
+      stream.on('data', chunk => {
+        hash.update(chunk);
+      });
+      stream.on('error', error => {
+        calculatedList.set(file, '');
+        callback();
+      });
+      stream.on('end', () => {
+        calculatedList.set(file, hash.digest('hex'));
+        callback();
+      });
+    }, error => {
+      resolve(calculatedList);
+    });
+  });
+}
+
+// REQUEST FOR S3 DOWNLOAD LINK OF A FILE
+getDownloadUrls = files => {
+  let query = qs.stringify({filenames: files}, {arrayFormat:'comma'});
+  let options = {
+    method: 'GET',
+    uri: djangoUrl + `/game-files/download?${query}`,
```

Comment:
is this dangerous to append the query like so? `/game-files/download?${query}`

Reply:
Not really, the http request module we use, uses the same qs module and appends query exactly like this. However, the version in the request module's dependency is older and does not support stringifying array with comma separators. That's why I had to include the new version and append query manually.